doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_ironside.rst

```diff
@@ -7,22 +7,161 @@ IronSide Secure Element
    :local:
    :depth: 2
 
-The IronSide Secure Element (|ISE|) is a firmware for the :ref:`Secure Domain <ug_nrf54h20_secure_domain>` of the nRF54H20 SoC that provides security features based on the :ref:`PSA Crypto API <ug_psa_certified_api_overview_crypto>`, part of the :ref:`PSA Certified Security Framework <ug_psa_certified_api_overview>`.
+The IronSide Secure Element (|ISE|) is a firmware for the :ref:`Secure Domain <ug_nrf54h20_secure_domain>` of the nRF54H20 SoC that provides security features based on the :ref:`PSA Certified Security Framework <ug_psa_certified_api_overview>`.
 
 |ISE| provides the following features:
 
 * Global memory configuration
 * Peripheral configuration (through UICR.PERIPHCONF)
 * Boot commands (ERASEALL, DEBUGWAIT)
+* An alternative boot path with a Secondary firmware
 * CPUCONF service
 * Update service
 * PSA Crypto service - see also :ref:`ug_crypto_architecture_implementation_standards_ironside`
+* PSA Internal Trusted Storage service
 
-Distribution
-************
+.. _ug_nrf54h20_ironside_se_programming:
 
-The |ISE| is provided as a precompiled binary, which is part of the nRF54H20 SoC bundle and is provided independently from the |NCS| release cycle.
-For more information, see :ref:`abi_compatibility`.
+Programming |ISE| on the nRF54H20 SoC
+*************************************
+
+|ISE| is included in the nRF54H20 SoC binaries.
+The nRF54H20 SoC binaries are released independently from the |NCS| release cycle and are bundled in a ZIP archive that contains the following components:
+
+* *IronSide SE update firmware* (:file:`ironside_se_update.hex`) - The main |ISE| firmware
+* *IronSide SE Recovery update firmware* (:file:`ironside_se_recovery_update.hex`) - The recovery firmware
+* The update application (:file:`update_application.hex`) - The application firmware used to trigger the update process
+* Additional metadata and manifest files required for the update process
+
+The bundle ZIP file follows the naming convention :file:`<soc>_soc_binaries_v<version>.zip`.
+
+For more information on the nRF54H20 SoC binaries, see :ref:`nRF54H20 SoC binaries<abi_compatibility>`.
+For instructions on how to program the nRF54H20 SoC binaries, see :ref:`ug_nrf54h20_SoC_binaries`.
+
+By default, the nRF54H20 SoC uses the following memory and access configurations:
+
+* *MRAMC configuration*: MRAM operates in *Direct Write mode*.
+* *MPC configuration*: All memory not reserved by Nordic firmware is accessible with read, write, and execute (RWX) permissions by any domain.
+* *TAMPC configuration*: The Access Port (AP) for the application core is enabled and available, allowing direct programming of all the memory not reserved by Nordic firmware in the default configuration.
```

Comment:
```suggestion
* MRAMC configuration - MRAM operates in *Direct Write mode*.
* MPC configuration - All memory not reserved by Nordic firmware is accessible with read, write, and execute (RWX) permissions by any domain.
* TAMPC configuration - The access port (AP) for the application core is enabled and available, allowing direct programming of all the memory not reserved by Nordic firmware in the default configuration.
```
Access port is not capitalized, see for example [nRF52840 docs](https://docs.nordicsemi.com/bundle/ps_nrf52840/page/dif.html#ariaid-title7)